docs/examples/event-processing.md

```diff
@@ -0,0 +1,211 @@
+    {"user_id":"a99", "email":"employee@bytewax.io"}
+    ]
+
+event_types = [
+    "click",
+    "post",
+    "comment",
+    "login"
+]
+
+try:
+    for i in range(500):
+        event = users[randint(0,4)]
+        event['type'] = event_types[randint(0,3)]
+        event_ = json.dumps(event).encode()
+        producer.send(input_topic_name, value=event_)
+        sleep(0.1)
+    print(f"input topic {input_topic_name} populated successfully")
+except KafkaError:
+    print("A kafka error occurred")
+```
+
+Now run `fake_events.py` to load our topic up with some event data.
+
+```python doctest:SKIP
+python fake_events.py
+```
+
+Dataflow
+-----------
+
+The dataflow will start with us defining the input. In this case, consuming from a Redpanda topic.
+
+```python doctest:SKIP
+
+​​from bytewax.dataflow import Dataflow
+from bytewax.inputs import KafkaInputConfig 
+
+flow = Dataflow()
+flow.input("inp", KafkaInputConfig(brokers=["localhost:9092"], topic="web_events"))
+```
+
+At a high-level, dataflow programming is a programming paradigm where program execution is conceptualized as data flowing through a series of operator-based steps. Operators like `map` and `filter` are the processing primitives of bytewax. Each of them gives you a “shape” of data transformation, and you give them regular Python functions to customize them to a specific task you need. See the documentation for a list of the [available operators](https://docs.bytewax.io/apidocs/bytewax.dataflow#bytewax.dataflow.Dataflow)
+
+```python doctest:SKIP
+import json
+def deserialize(key_bytes__payload_bytes):
+    key_bytes, payload_bytes = key_bytes__payload_bytes
+    key = json.loads(key_bytes) if key_bytes else None
+    event_data = json.loads(payload_bytes) if payload_bytes else None
+    return event_data["user_id"], event_data
+
+def anonymize_email(user_id__event_data):
+    user_id, event_data = user_id__event_data
+    event_data["email"] = "@".join(["******", event_data["email"].split("@")[-1]])
+    return user_id, event_data
+
+def remove_bytewax(user_id__event_data):
+    user_id, event_data = user_id__event_data
+    return "bytewax" not in event_data["email"]
+
+flow.map(deserialize)
+flow.map(anonymize_email)
+flow.filter(remove_bytewax)
+```
+
+Bytewax is a stateful stream processor, which means that you can do things like aggregations and windowing. With Bytewax, state is stored in memory on the workers by default, but can also be persisted with different [state recovery mechanisms](https://docs.bytewax.io/apidocs/bytewax.recovery) for failure recovery. There are different stateful operators available like `reduce`, `stateful_map` and `fold_window`. The complete list can be found in the [API documentation for all operators](https://docs.bytewax.io/apidocs/bytewax.dataflow). Below we use the `fold_window` operator with a tumbling window based on system time to gather events and calculate the number of times different events happen per user.
```

Comment:
this isn't really an answer we need at this moment but i keep wondering if we want to move toward "processing time" or perhaps "system processing time" over "system time"